docs/node-operators/migration-guides/upgrade-to-0710.md

```diff
@@ -0,0 +1,390 @@
+
+Please follow [the standard protocol upgrade documentation](../how-to/upgrade-network.md). This document has only key points and suggested changes for the vega setup.
+
+## The data node configuration changes
+
+Once you have `voted` for the protocol upgrade, but `before the protocol upgrade happens`, please update the following sections in the data-node config:
+
+### 1. The Gateway Port settings
+
+We have merged the `GraphQL` and the `REST` APIs into a single port. Now you must define the `Port` and `IP` in the `[Gateway]` section instead of its child. Remove the `Port` and `IP` from the `[Gateway.GraphQL]` and `[Gateway.REST]` sections.
+
+```diff
+  [Gateway]
++   Port = 443
++   IP = "0.0.0.0"
+    [Gateway.GraphQL]
+-     Port = 3008
+-     IP = "0.0.0.0" 
+      ...
+    [Gateway.REST]
+-     Port = 3008
+-     IP = "0.0.0.0" 
+      ...
+```
+
+### 2. Enable the TLS on the data node API
+
+The data node can request the TLS certificate for you automatically. The only requirement is to use port 443 for the Gateway component and has this port open to the public internet. You can still use a custom port and some software like Apache or Nginx and proxy pass requests to the data node, but in this case, you are responsible for generating and renewing the certificate.
+
+Below is an example certificate for the data node auto TLS:
+
+```toml
+[Gateway]
+  Port = 443
+  IP = "0.0.0.0"
+  Level = "Info"
+  Timeout = "5s"
+  SubscriptionRetries = 3
+  GraphQLPlaygroundEnabled = true
+  MaxSubscriptionPerClient = 250
+  HTTPSEnabled = true
+  AutoCertDomain = "api.vega.example.com"
+  CertificateFile = ""
+  KeyFile = ""
+  [Gateway.GraphQL]
+    Enabled = true
+    ComplexityLimit = 0
+    Endpoint = "/graphql"
+  [Gateway.REST]
+    Enabled = true
+    APMEnabled = true
+  [Gateway.CORS]
+    AllowedOrigins = ["*"]
+    MaxAge = 7200
+...
+```
+
+If the `api.vega.example.com` domain points to your data node, it will obtain the certificate from the Let's Encrypt.
+
+:::note All the differences
+To see all the differences, we recommend generating a new configuration and compare with your configuration on the node. The procedure is described [here](./upgrade-node.md#16-update-data-node-config)
+:::
+
+## The vegavisor configuration change.
+
+You can still use the old binary for the vegavisor and skip preparing the new configuration, but We recommend upgrading and preparing a new configuration for the vegavisor.
+
+
+:::node Documentation
+To read documentation for the vegavisor config visit [the vegavisor documentation page](https://github.com/vegaprotocol/vega/blob/develop/visor/visor-config.md)
+
+The excample vegavisor config is:
+
+```toml
+maxNumberOfFirstConnectionRetries = 43200
+maxNumberOfRestarts = 3
+restartsDelaySeconds = 5
+stopDelaySeconds = 0
```

Comment:
You can highlight this one as a new one, and tell that this is useful for Sentry Node.

Reply:
fixed